geomstats/geometry/sasaki_metric.py

```diff
@@ -0,0 +1,238 @@
+"""Class for the Sasaki metric.
+
+A class implementing the Sasaki metric: The natural metric on the tangent bundle TM
+of a Riemannian manifold M.
+
+Lead authors: E. Nava-Yazdani, F. Ambellan, M. Hanik and C. von Tycowicz.
+"""
+import os
+from joblib import Parallel, delayed
+
+import geomstats.backend as gs
+from geomstats.geometry.riemannian_metric import RiemannianMetric
+
+
+def _gradient_descent(x_ini, grad, exp, lrate=0.1, max_iter=100, tol=1e-6):
+    """Apply a gradient descent until max_iter or a given tolerance is reached."""
+    L = len(x_ini)
+    x = x_ini
+    for _ in range(max_iter):
+        grad_x = grad(x)
+        grad_norm = gs.linalg.norm(grad_x)
+        if grad_norm < tol:
+            break
+        grad_x = -lrate * grad_x
+        for j in range(L):
+            x[j] = exp(grad_x[j], x[j])
```

Comment:
Can be vectorized: `self.exp = exp(grad_x, x)`.

Reply:
Ok